src/runtime/internal/dom.ts

```diff
@@ -180,7 +216,7 @@ export function claim_text(nodes, data) {
 		const node = nodes[i];
 		if (node.nodeType === 3) {
 			node.data = '' + data;
-			return nodes.splice(i, 1)[0];
+			return nodes.shift();
```

Comment:
this line is weird.

you found a text element in the middle of the nodes array, but you claim the first element ?

Reply:
Hm! When I revert this specific change on this line I'm no longer seeing the bug in #6274. I have no idea what else is affected though.

Reply:
This could only worked with `shift()` if _all_ `nodes` were text nodes